### PR TITLE
devicetree: ATMEL SAM0: Convert i2c & spi to new devicetree

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by nandojve** [comment]
SPI tested using SAMR21 running samples/net/echo_client with/without DMA.

**Comment by nandojve** [comment]
Tested on SAMR21 with/without DMA
```
muart:~$ *** Booting Zephyr OS build zephyr-v2.2.0-1911-g88259bbbe6ec  ***


UI-64: 0x263ffadbc72b89c9
```

File: dts/bindings/spi/atmel,sam0-spi.yaml [hunk 1]
```diff
@@ -27,12 +27,19 @@ properties:
...
-      required: false
-      description: Transmit DMA channel
+    dmas:
+      description: |
+        Optional TX & RX dma specifiers.  Each specifier will have a phandle
+        reference to the dmac controller, the channel number, and peripheral
+        trigger source.
+
+        For example dmas for TX, RX on SERCOM3
+           dmas = <&dmac 0 0xb>, <&dmac 0 0xa>;
```
Comment (nandojve) [inline dts/bindings/spi/atmel,sam0-spi.yaml:37]: The channel should be different on SPI.
dmas = <&dmac **1** 0xb>, <&dmac **0** 0xa>; maybe?

**Review by nandojve — CHANGES_REQUESTED** [review]
Small typo on docs.

**Comment by galak** [comment]
> muart:~$ *** Booting Zephyr OS build zephyr-v2.2.0-1911-g88259bbbe6ec  ***
> 
> 
> UI-64: 0x263ffadbc72b89c9

Can you see if its the I2C or SPI change or both?

- k

File: dts/bindings/spi/atmel,sam0-spi.yaml (hunk 1 above)
Comment (galak) [reply dts/bindings/spi/atmel,sam0-spi.yaml:37]: Done

**Comment by nandojve** [comment]
> > muart:~$ *** Booting Zephyr OS build zephyr-v2.2.0-1911-g88259bbbe6ec  ***
> > UI-64: 0x263ffadbc72b89c9
> 
> Can you see if its the I2C or SPI change or both?
> 
> * k

Ahh...I2C above is, SPI already commented yesterday.
Both protocols worked well.

**Review by nandojve — APPROVED** [review]

**Pull request merged**